### PR TITLE
VPN-4493 - Use cmake iOS build on XCode cloud

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

**Review by bakulf — COMMENTED** [review]
What's the reason to replace mozillavpn with the variable?

File: ci_scripts/ci_post_clone.sh [hunk 1]
```diff
@@ -68,9 +31,48 @@ APP_ID_IOS = org.mozilla.ios.FirefoxVPN
...
+# In previous iterations "Mozilla VPN" didn't use to have a space in it,
+# to avoid the pains of migrations we will just rename here. 
+# This is harmless, but still very hacky.
+WORKSPACE_ROOT="$( cd "$(dirname "$0")/.." ; pwd -P )"
+sed -i.bak -E \
+    -e "s/Mozilla VPN/MozillaVPN/" \
+    "${WORKSPACE_ROOT}/CMakeLists.txt"
+sed -i.bak -E \
+    -e "s/MAIN_TARGET mozillavpn/MAIN_TARGET MozillaVPN/" \
+    "${WORKSPACE_ROOT}/CMakeLists.txt"
```
Comment (brizental) [inline ci_scripts/ci_post_clone.sh:72]: @bakulf this is the reason for replacing `mozillavpn` with the variable. I wanted to make this last replace easier.

Comment (bakulf) [reply ci_scripts/ci_post_clone.sh:71]: This can be achieved  using cmake flags.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -43,30 +45,30 @@ endif()
 include(shared/sources.cmake)
 include(apps/vpn/cmake/sources.cmake)
```
Comment (bakulf) [inline src/CMakeLists.txt:47]: This can be called as you like.

File: src/apps/vpn/cmake/windows.cmake [hunk 3]
```diff
@@ -4,25 +4,25 @@
 
 add_definitions(-DWIN32_LEAN_AND_MEAN)
 
-set_target_properties(mozillavpn PROPERTIES
+set_target_properties(${MAIN_TARGET} PROPERTIES
```
Comment (bakulf) [inline src/apps/vpn/cmake/windows.cmake:7]: This property values is what you want to change. Not the rest.

**Review by bakulf — CHANGES_REQUESTED** [review]

File: ci_scripts/ci_post_clone.sh (hunk 1 above)
Comment (brizental) [reply ci_scripts/ci_post_clone.sh]: I have attempted to achieve this by changing multiple variables listed in https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html and https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#target-properties without success in either changing the project name or the mozillavpn target name. 

The objective here is to generate an `.xcodeproj` file with the name `MozillaVPN.xcodeproj` and the main scheme to be also `MozillaVPN`. @oskirby @bakulf please let me know if you know how to do this.

File: src/CMakeLists.txt (hunk 2 above)
Comment (brizental) [reply src/CMakeLists.txt:45]: @bakulf I don't understand what you mean by this comment. Would you mind expanding?

File: src/apps/vpn/cmake/windows.cmake (hunk 3 above)
Comment (brizental) [reply src/apps/vpn/cmake/windows.cmake:7]: Pleaes see my previous comment. I have not had success on changing the XCode scheme name just by changing a target property.

**Comment by oskirby** [comment]
Closing this as it has been superseeded by #6658.
I cannot thank you enough for your contributions on this @brizental, thank you so much!

**Pull request closed without merging**